tests/client/test_client_multi_search_meilisearch.py

```diff
@@ -32,3 +33,49 @@ def test_multi_search_on_no_index(client):
     """Tests multi-search on a non existing index."""
     with pytest.raises(MeilisearchApiError):
         client.multi_search([{"indexUid": "indexDoesNotExist", "q": ""}])
+
+
+def test_multi_search_with_no_value_in_federation(client, empty_index,
+                                                  index_with_documents):
+    """Tests multi-search with federation, but no value"""
+    index_with_documents()
+    empty_index("indexB")
+    response = client.multi_search(
+        [{"indexUid": INDEX_UID, "q": ""}, {"indexUid": "indexB", "q": ""}], {}
+    )
+    assert "results" not in response
+    assert len(response["hits"]) > 0
+    assert "_federation" in response["hits"][0]
+    assert response["limit"] == 20
+    assert response["offset"] == 0
+
+
+def test_multi_search_with_offset_and_limit_in_federation(
+        client,
+        index_with_documents):
+    """Tests multi-search with federation, with offset and limit value"""
+    index_with_documents()
+    response = client.multi_search([{"indexUid": INDEX_UID, "q": ""}],
+                                   {"offset": 2, "limit": 2})
+
+    assert "results" not in response
+    assert len(response["hits"]) == 2
+    assert "_federation" in response["hits"][0]
+    assert response["limit"] == 2
+    assert response["offset"] == 2
+
+
+def test_multi_search_with_federation_options(client, index_with_documents):
+    """Tests multi-search with federation, with federation options"""
+    index_with_documents()
+    response = client.multi_search(
+        [{"indexUid": INDEX_UID, "q": "", 
+          "federationOptions": {"weight": 0.99}}], {"limit": 2})
+
+    assert "results" not in response
+    assert response["hits"][0]["_federation"]["indexUid"] == INDEX_UID
+    assert isinstance(response["hits"], list)
+    assert response["hits"][0]["_federation"]["weightedRankingScore"] >= 0.99
+    assert len(response["hits"]) == 2
```

Comment:
`assert isinstance(response["hits"], list)` would never run because an exception would be thrown and there is the potential for the same with `assert len(response["hits"]) == 3`. Rearranging things will fix that.

```suggestion
    assert isinstance(response["hits"], list)
    assert len(response["hits"]) == 2
    assert response["hits"][0]["_federation"]["indexUid"] == INDEX_UID
    assert response["hits"][0]["_federation"]["weightedRankingScore"] >= 0.99
```

Reply:
Yes, oh yes. Right. Thank you so much.